src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -32,29 +32,35 @@ namespace SonarScanner.MSBuild.PreProcessor
 {
     public class WebClientDownloader : IDownloader
     {
+        // This is a temporary solution untill we upgrade to .net framework 4.8.
+        private const SecurityProtocolType Tls13 = (SecurityProtocolType)12288;
+        private const SecurityProtocolType SystemDefault = 0;
+
         private readonly ILogger logger;
         private readonly HttpClient client;
 
         public WebClientDownloader(string userName, string password, ILogger logger, string clientCertPath = null, string clientCertPassword = null)
         {
-            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            if (password == null)
+            if (ServicePointManager.SecurityProtocol != SystemDefault)
             {
-                password = "";
+                ServicePointManager.SecurityProtocol = Tls13 | SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
```

Comment:
do we need `SecurityProtocolType.Tls11 | SecurityProtocolType.Tls` at all?

SQ/SC should be ok with min ver Tls12

Reply:
Do we know what's the time evolution of minimal supported version on SQ side? Because the latest scanner can be used with old SQ in locked environment where users might be stuck on old TLS

Reply:
If I understand it correctly it could be a breaking change for users who for some reason run the scanner on machines that don't have TLS1.2 available. Together with @tom-howlett-sonarsource we decided not to introduce breaking changes at least not without prior warnings so that is why those versions are still enabled.

Reply:
ok

Reply:
Yes that part is only executed for older .NET FW versions.

Reply:
ok, than you should mark them as won't fix in SC with an explanation message